src/check_ast.rs

```diff
@@ -1946,71 +1956,100 @@ impl<'a> Checker<'a> {
     }
 
     fn handle_node_load(&mut self, expr: &Expr) {
-        if let ExprKind::Name { id, .. } = &expr.node {
-            let scope_id =
-                self.scopes[*(self.scope_stack.last().expect("No current scope found."))].id;
-
-            let mut first_iter = true;
-            let mut in_generator = false;
-            let mut import_starred = false;
-            for scope_index in self.scope_stack.iter().rev() {
-                let scope = &mut self.scopes[*scope_index];
-                if matches!(scope.kind, ScopeKind::Class(_)) {
-                    if id == "__class__" {
-                        return;
-                    } else if !first_iter && !in_generator {
-                        continue;
-                    }
-                }
-
-                if try_mark_used(scope, scope_id, id, expr) {
-                    return;
+        // TODO: Cleanup
+        // Some modules can be accessed from the their parent module, so here we
+        // create all possible access patterns. That means it's possible we do
+        // not find all unused imports, but at least we're not reporting
+        // false-positives. One example is `os.path`, which can be accessed when
+        // importing either `os` or `os.path`.
+        let ids = match &expr.node {
+            ExprKind::Name { id, .. } => vec![(id.clone(), expr.clone())],
+            ExprKind::Attribute { value, .. } => {
+                let path = compose_call_path(expr).unwrap();
+                let segments: Vec<&str> = path.split('.').collect();
+                if segments.len() > 1 {
```

Comment:
I think this _always_ has to be true, right?

Reply:
Yes, you are right. Resolving this because the current code doesn't care anymore.